src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs

```diff
@@ -1524,13 +1524,20 @@ private void ProcessErrorThread(object state)
                 while (true)
                 {
                     string error = reader.ReadLine();
-                    if (!string.IsNullOrEmpty(error) && (error.IndexOf("WARNING:", StringComparison.OrdinalIgnoreCase) < 0))
+                    if (!string.IsNullOrEmpty(error) && (error.IndexOf("WARNING:", StringComparison.OrdinalIgnoreCase) > -1))
+                    {
+                        // Handle as interactive warning message.
+                        Console.WriteLine(error);
+                    }
```

Comment:
The transport doesn't have any concept of "transport warnings", only transport errors that ends the connection.  I don't really want to add transport warnings to the PowerShell warning stream since it is not the same thing.  Ditto for the output stream.  It might make sense to make them non-terminating errors, but again they apply to the transport and not running script so could be confusing.  Also right now I am not really sure whether we need to surface ssh client warnings.  So far the only warning I have received is for invalid key file path, but we now have a check for it and so it is not really needed.  So for now I think the best thing to do is surface them via console and see how necessary they may be.  We can always revisit later.

---
In reply to: [78627479](https://github.com/PowerShell/PowerShell/pull/2249#discussion_r78627479) [](ancestors = 78627479)